src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EditProceedingsAndCostsSectionController.java

```diff
@@ -25,63 +61,679 @@
 @Slf4j
+@SessionAttributes(value = {
+    APPLICATION,
+    APPLICATION_PROCEEDINGS,
+    APPLICATION_COSTS,
+    APPLICATION_PRIOR_AUTHORITIES,
+    CURRENT_PROCEEDING,
+    PROCEEDING_FLOW_FORM_DATA,
+    PROCEEDING_FLOW_FORM_DATA_OLD,
+    PROCEEDING_SCOPE_LIMITATIONS})
 public class EditProceedingsAndCostsSectionController {
 
   private final ApplicationService applicationService;
 
+  private final LookupService lookupService;
+
+  private final ProceedingMatterTypeDetailsValidator matterTypeValidator;
+  private final ProceedingTypeDetailsValidator proceedingTypeValidator;
+  private final ProceedingFurtherDetailsValidator furtherDetailsValidator;
+
+  private final ProceedingAndCostsMapper proceedingAndCostsMapper;
+
+  private static final String ACTION_EDIT = "edit";
+  private static final String ACTION_ADD = "add";
+  private static final String STATUS_DISPLAY_VALUE_ADDED = "Added";
+  private static final String STATUS_DISPLAY_VALUE_UPDATED = "Updated";
+  private static final String STATUS_DRAFT = "Draft";
+
   /**
    * Handles the GET request to fetch and display the proceedings and costs for a specific
    * application.
    *
    * @param applicationId The id of the application, retrieved from the session.
    * @param model The Model object to add attributes to for the view.
-   * @param session The HttpSession object to add attributes to for the session.
    * @return The name of the view to be rendered.
    */
   @GetMapping("/application/summary/proceedings-and-costs")
   public String proceedingsAndCosts(
       @SessionAttribute(APPLICATION_ID) final String applicationId,
-      final Model model,
-      final HttpSession session) {
+      @SessionAttribute(USER_DETAILS) final UserDetail user,
+      final Model model) {
 
-    final Mono<ResultsDisplay<Proceeding>> proceedingsMono =
-        applicationService.getProceedings(applicationId);
-    final Mono<CostStructure> costsMono =
-        applicationService.getCosts(applicationId);
-    final Mono<ResultsDisplay<PriorAuthority>> priorAuthoritiesMono =
-        applicationService.getPriorAuthorities(applicationId);
-
-    return Mono.zip(proceedingsMono,
-            costsMono,
-            priorAuthoritiesMono)
-        .map(tuple -> {
-          model.addAttribute("proceedings", tuple.getT1());
-          session.setAttribute("proceedings", tuple.getT1());
-          model.addAttribute("costs", tuple.getT2());
-          session.setAttribute("costs", tuple.getT2());
-          model.addAttribute("priorAuthorities", tuple.getT3());
-          session.setAttribute("priorAuthorities", tuple.getT3());
-
-          return "application/summary/proceedings-and-costs-section";
-        }).block();
+    final ApplicationDetail application =
+        applicationService.getApplication(applicationId).block();
+
+    applicationService.prepareProceedingSummary(applicationId, application, user);
+    model.addAttribute(APPLICATION, application);
+
+    if (application != null) {
```

Comment:
If the application was null, would that not be an error condition? To be thrown earlier
Optional.ofNullable(applicationService.getApplication....).block()).orElseThrow(....)

Reply:
Amended